android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java

```diff
@@ -1187,6 +1342,37 @@ public void testPagedPull() throws InterruptedException {
     }
 
     @Test
+    public void testPagedPullBlocking() throws InterruptedException, IOException {
+        DataStore<Person> store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.CACHE, client);
+        client.getSyncManager().clear(Person.COLLECTION);
+
+        cleanBackendDataStore(store);
+
+        // Arrange
+        ArrayList<Person> persons = new ArrayList<>();
+        Person alvin = createPerson("Alvin");
+        Person simon = createPerson("Simon");
+        Person theodore = createPerson("Theodore");
+        save(store, alvin);
+        save(store, simon);
+        save(store, theodore);
+        long cacheSizeBefore = getCacheSize(StoreType.CACHE);
+        assertTrue(cacheSizeBefore == 3);
+        client.getCacheManager().getCache(Person.COLLECTION, Person.class, StoreType.CACHE.ttl).clear();
+        long cacheSizeBetween = getCacheSize(StoreType.CACHE);
+        assertTrue(cacheSizeBetween == 0);
+
+        // Act
+        store.setAutoPagination(true);
+        List<Person> pullResults = store.pullBlocking(null);
```

Comment:
Even though `autoPagination` is set to `true`, would this paginate on 3 items? Is there a page size that needs to be set?

Reply:
In my manual testing of the app, I set the page size to 2 items, to properly test the auto-pagination. In this test, though, this will just verify that this works on a single page. I could make the page size configurable, but I was not sure if it is a good idea to make this available, just for testing purposes.  Should I make the page size configurable?

Reply:
Hmm I see what you mean. It'll be nice to actually test auto-pagination. The page size doesn't need to be publicly available, but see if there's a way we can expose it for the test, or otherwise make the test retrieve multiple pages.